File: AngularUi/src/app/validators/student-validator.ts
import { AsyncValidator } from 'fluentvalidation-ts';
import { Student } from '../models/student';

export class StudentValidator extends AsyncValidator<Student> {
  constructor() {
    super();

    this.ruleFor('id')
      .notEmpty()
      .withMessage('Id is required')
      .matches(/^SCH[0-9]{4}$/)
      .withMessage('Id must be in the format SCH0000');

    this.ruleFor('firstName')
      .notEmpty()
      .withMessage('First name is required')
      .minLength(2)
      .withMessage('First name must be at least 2 characters long')
      .maxLength(30)
      .withMessage('First name must be at most 30 characters long');

    this.ruleFor('lastName')
      .notEmpty()
      .withMessage('Last name is required')
      .minLength(2)
      .withMessage('First name must be at least 2 characters long')
      .maxLength(30)
      .withMessage('First name must be at most 30 characters long');

    this.ruleFor('contactNumber')
      .notEmpty()
      .withMessage('Contact number is required')
      .matches(/^[0-9]{10}$/)
      .withMessage('Contact number must be 10 digits long');

    this.ruleFor('emailAddress')
      .notEmpty()
      .withMessage('Email address is required')
      .emailAddress()
      .withMessage('Email address must be a valid email address');

  }
}

File: WebApi/Domain/Validations/StudentValidator.cs
using Domain.Students;
using FluentValidation;

namespace Domain.Validations
{
    public class StudentValidator : AbstractValidator<Student>
    {
        public StudentValidator()
        {

        RuleFor(x => x.FirstName)
        .NotEmpty()
        .WithMessage("First name is required")
        .MinimumLength(2)
        .WithMessage("First name must be at least 2 characters long")
        .MaximumLength(30)
        .WithMessage("First name must be at most 30 characters long");

        RuleFor(x => x.LastName)
        .NotEmpty()
        .WithMessage("Last name is required")
        .MinimumLength(2)
        .WithMessage("Last name must be at least 2 characters long")
        .MaximumLength(30)
        .WithMessage("Last name must be at most 30 characters long");

        RuleFor(x => x.ContactNumber)
        .NotEmpty()
        .WithMessage("Contact number is required")
        .Matches(@"^\d{10}$")
        .WithMessage("Contact number must be 10 digits long");

        RuleFor(x => x.EmailAddress)
        .NotEmpty()
        .WithMessage("Email address is required")
        .EmailAddress()
        .WithMessage("Email address must be a valid email address");

        }
    }
}
